### ISSUE TITLE
Right to Left Layout Problem

### ISSUE THREAD

**Issue by amiraliakbari** [nextcloud/talk-android#405]
When I run Nextcloud Talk, it crashes with: `java.lang.IllegalStateException: Circular dependencies cannot exist in RelativeLayout`. I have tested the F-Droid build and compiling the latest commit.

The phone I am testing on shows applications in right to left layout by default. By adding `android:supportsRtl="false"` to manifest the problem is resolved, so I think the cause is RTL layout.

For reference, the full exception log is:
```
java.lang.IllegalStateException: Circular dependencies cannot exist in RelativeLayout
       at android.widget.RelativeLayout$DependencyGraph.getSortedViews(RelativeLayout.java:1909)
       at android.widget.RelativeLayout.sortChildren(RelativeLayout.java:403)
       at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:410)
       at android.view.View.measure(View.java:19149)
       at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:825)
       at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:511)
       at android.view.View.measure(View.java:19149)
       at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:825)
       at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:511)
       at android.view.View.measure(View.java:19149)
       at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:825)
       at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:511)
       at android.view.View.measure(View.java:19149)
       at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:825)
       at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:511)
       at android.view.View.measure(View.java:19149)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6138)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:223)
       at studio.carbonylgroup.textfieldboxes.TextFieldBoxes.onMeasure(TextFieldBoxes.java:357)
       at android.view.View.measure(View.java:19149)
       at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:825)
       at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:511)
       at android.view.View.measure(View.java:19149)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6138)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:223)
       at android.view.View.measure(View.java:19149)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6138)
       at androidx.coordinatorlayout.widget.CoordinatorLayout.onMeasureChild(CoordinatorLayout.java:733)
       at com.google.android.material.appbar.HeaderScrollingViewBehavior.onMeasureChild(HeaderScrollingViewBehavior.java:95)
       at com.google.android.material.appbar.AppBarLayout$ScrollingViewBehavior.onMeasureChild(AppBarLayout.java:1556)
       at androidx.coordinatorlayout.widget.CoordinatorLayout.onMeasure(CoordinatorLayout.java:803)
       at android.view.View.measure(View.java:19149)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6138)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:223)
       at androidx.appcompat.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:143)
       at android.view.View.measure(View.java:19149)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6138)
       at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1723)
       at android.widget.LinearLayout.measureVertical(LinearLayout.java:788)
       at android.widget.LinearLayout.onMeasure(LinearLayout.java:648)
       at android.view.View.measure(View.java:19149)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6138)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:223)
       at android.view.View.measure(View.java:19149)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6138)
       at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1723)
       at android.widget.LinearLayout.measureVertical(LinearLayout.java:788)
       at android.widget.LinearLayout.onMeasure(LinearLayout.java:648)
       at android.view.View.measure(View.java:19149)
       at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6138)
       at android.widget.FrameLayout.onMeasure(FrameLayout.java:223)
       at com.android.internal.policy.PhoneWindow$DecorView.onMeasure(PhoneWindow.java:2700)
       at android.view.View.measure(View.java:19149)
       at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:2481)
       at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1440)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1694)
       at android.view.ViewRootImpl.doTravers
```

**Comment by mario** [comment]
Thanks for the report, I'll take a look.

**Comment by mario** [comment]
Ideally, I'll remove the complete library in the follow-up stable release. Thanks for finding this.

**Comment by nickvergessen** [comment]
Minimum solution includes:
- [ ] Align conversations in the conversations list to the right.
- [ ] The user’s own chat messages will be aligned to the left.
- [ ] Received chat messages will be aligned to the right.

**Comment by AndyScherzinger** [comment]
all minimum solution requirements are already implemented, added or improved via #1483
